support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/SupporterPlusSubcriptionBuilder.scala

```diff
@@ -49,6 +56,46 @@ class SupporterPlusSubcriptionBuilder(
       salesforceCaseId = salesforceCaseId,
     )
     subscribeItemBuilder.build(subscriptionData, state.salesForceContact, Some(state.paymentMethod), None)
+  }
+  def buildV2(state: SupporterPlusState, csrUsername: Option[String], salesforceCaseId: Option[String]) = {
+    val productRatePlanId =
+      validateRatePlan(supporterPlusRatePlanV2(state.product, environment), state.product.describe)
+    val contributionRatePlanChargeId =
+      if (state.product.billingPeriod == Monthly) config.v2.monthlyContributionChargeId
+      else config.v2.annualContributionChargeId
+    val todaysDate = dateGenerator.today
+
+    val contributionAmount = state.product.amount - getBaseProductPrice(productRatePlanId, state.product.currency)
+    val subscriptionData = subscribeItemBuilder.buildProductSubscription(
+      productRatePlanId = productRatePlanId,
+      ratePlanCharges = List(
+        RatePlanChargeData(
+          RatePlanChargeOverride(
+            contributionRatePlanChargeId,
+            price = contributionAmount, // Pass the amount of the cost which counts as a contribution into Zuora
```

Comment:
The other charge, the supporter plus subscription charge, inherits the price from the catalog automatically when the sub is created. We override the contribution charge amount because it is variable depending on what the user has chosen to pay.

Reply:
Ah of course – thanks for the explanation